lib/vdsm/virt/vm.py

```diff
@@ -1490,199 +1429,29 @@ def refresh_disk(self, vol_pdiv):
-        to refresh destination before source to be sure that the disk on the
-        destination won't be corrupted by resumed VM. We need to call it only
-        from the source VM.
-
-        Returns True if aforementioned situation happens and the disk on the
-        destination has to be refreshed.
-        """
-        return self._migrationSourceThread.needs_disk_refresh()
-
-    def after_volume_extension(self, volInfo):
-        callback = None
-        error = None
-        try:
-            callback = volInfo["callback"]
-            clock = volInfo["clock"]
-            clock.stop("extend-volume")
-
-            if self.should_refresh_destination_volume():
-                with clock.run("refresh-destination-volume"):
-                    self._refresh_destination_volume(volInfo)
-
-            with clock.run("refresh-volume"):
-                self.refresh_drive_volume(volInfo)
-
-            # Check if the extension succeeded.  On failure an exception is
-            # raised.
-            # TODO: Report failure to the engine.
-            volSize = self.__verifyVolumeExtension(volInfo)
-
-            # This was a volume extension or replica and volume extension.
-            clock.stop("total")
-            self.log.info("Extend volume %s completed %s",
-                          volInfo["volumeID"], clock)
-
-            # Only update apparentsize and truesize if we've resized the leaf
-            if not volInfo['internal']:
-                drive = lookup.drive_by_name(
-                    self.getDiskDevices()[:], volInfo['name'])
-                self._update_drive_volume_size(drive, volSize)
-
-            self._resume_if_needed()
-
-        except exception.DiskRefreshNotSupported as e:
-            self.log.warning(
-                "Migration destination host does not support "
-                "extending disk during migration, disabling disk "
-                "extension during migration")
-            self.drive_monitor.disable()
-            error = e
-        except virdomain.NotConnectedError as e:
-            self.log.debug("VM not running, aborting extend completion")
-            error = e
-        finally:
-            if callback:
-                callback(error=sys.exc_info()[1] or error)
-
-    def _update_drive_volume_size(self, drive, volsize):
+    def extend_volume_completed(self):
         """
-        Updates drive's apparentsize and truesize, and set a new block
-        threshold based on the new size.
+        Called when extending a thin volume completed successfuly.
 
-        Arguments:
-            drive (virt.vmdevices.storage.Drive): The drive object using the
-                resized volume.
-            volsize (virt.vm.VolumeSize): new volume size tuple
+        If a guest writes data too fast, the VM may pause with ENOSPC before
+        the volume extension was completed. If the VM is paused, we try to
+        resume it incase it was paused because of a slow extend.
```

Comment:
- ... in case ...
- I don't think there is a noun "extend" in English (possible alternatives: extent, extension, extending, ...).

Reply:
How about:

    resume in case it was paused because of a slow extend operation.

Reply:
Not sure about native speakers but good enough for me. :-)

Reply:
Native speakers are invited to contribute :-)

Reply:
I changed the text to use "volume extension".